packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/pages/delete/pages/[...slug].astro

```diff
@@ -1,72 +1,90 @@
 
-const ConfigArray = await db.select().from(SiteConfig);
-const pageArrayed = await db.select().from(Pages).where(eq(Pages.slug, slug));
-
-const page = pageArrayed[0];
+const contextConfig: SiteConfigResponse = await db.select().from(SiteConfig).where(eq(SiteConfig.id, 1)).get();
+const page: pageDataReponse = await db.select().from(PageData).where(eq(PageData.slug, slug)).get();
 
 if (!slug || !page) {
-	return Astro.redirect(await urlGenFactory(true, undefined, dashboardRouteOverride));
+	return Astro.redirect(dashboardIndex);
 }
 
-const contextConfig = ConfigArray.pop();
-
-const locals = Astro.locals as Locals;
-const { permissionLevel } = await authHelper(locals);
+const deleteURL = getDeleteRoute(slug);
+const { permissionLevel } = await authHelper(Astro.locals as Locals);
 
 // If the user is not logged in, redirect to the login page
 if ( !developerConfig.testingAndDemoMode ) {
     if ( permissionLevel === "unknown" ) {
         console.log('User is not logged in. Redirecting to login page.');
-        return Astro.redirect(await urlGenFactory(true, "login", dashboardRouteOverride));
+        return Astro.redirect(loginURL);
     }
     if ( permissionLevel !== 'admin' && permissionLevel !== 'editor' ) {
         console.log('User is not an admin or editor. Redirecting to profile page.');
-        return Astro.redirect(await urlGenFactory(true, "profile", dashboardRouteOverride));
+        return Astro.redirect(userProfile);
     }
 }
 
 let showSuccessToast = false;
 let showErrorToast = false;
-let showSlugErrorToast = false;
+// let showSlugErrorToast = false;
 
 if (Astro.request.method === 'POST' && !developerConfig.testingAndDemoMode) {
     try {
         const data = await Astro.request.formData();
         const slug = data.get('slug');
         if (slug === page.slug) {
             if ( slug === "index" || slug === "about" ) {
+                showErrorToast = true;
                 return new Response('Error: Cannot delete the index or about page.', { status: 500 })
             }
+
             await db
-                .delete(Pages)
-                .where(eq(Pages.slug, page.slug))
+                .delete(PageData)
+                .where(eq(PageData.slug, page.slug as string))
                 .then(() => {
                     showSuccessToast = true;
                 })
-                .catch(() => {
+                // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                .catch((e: any) => {
+                    console.error(e);
                     showErrorToast = true;
                 });
-        } else {
-            showSlugErrorToast = true;
-        }
+
+            await db
+                .delete(PageContent)
+                .where(eq(PageContent.contentId, page.id))
+                .then(() => {
+                    showSuccessToast = true;
+                })
+                // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                .catch((e: any) => {
```

Comment:
Eventually it's probably worth thinking of some global error catching/handling/displaying mechanism, but this is fine for now.

Reply:
Logic handling for these pages in regards to editing the database is to be moved to an API Endpoint for better response handling and toasting (see the start of this in the other pages, such as auth routes)

Reply:
We could use the [effect pattern](https://effect.website/docs/why-effect#the-effect-pattern)

Reply:
> We could use the [effect pattern](https://effect.website/docs/why-effect#the-effect-pattern)

interesting... never seen that, but honestly, with Astro's API endpoints, i think those should be enough being able to return proper responses.  (the current plan that I've been building towards since i have to move all the logic anyway so that its easier to interact from different dashboard pages)